core/modules/qproc/testQueryAnaAggregation.cc

```diff
@@ -0,0 +1,101 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+ /**
+  * @file
+  *
+  * @brief Test C++ parsing and query analysis logic for select expressions
+  * with an "ORDER BY" clause.
+  *
+  *
+  * @author Fabrice Jammes, IN2P3/SLAC
+  */
+
+// System headers
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <sstream>
+#include <string>
+
+// Third-party headers
+#include "boost/algorithm/string.hpp"
+#include "boost/format.hpp"
+
+// Boost unit test header
+#define BOOST_TEST_MODULE QueryAnaAggregation
+#include "boost/test/included/unit_test.hpp"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "qproc/testQueryAna.h"
+#include "qproc/QuerySession.h"
+#include "query/QueryContext.h"
+
+using lsst::qserv::query::SelectStmt;
+using lsst::qserv::query::QueryContext;
+
+////////////////////////////////////////////////////////////////////////
+// CppParser basic tests
+////////////////////////////////////////////////////////////////////////
+BOOST_FIXTURE_TEST_SUITE(Aggregate, ParserFixture)
+
+BOOST_AUTO_TEST_CASE(Aggregate) {
+    std::string stmt = "select sum(pm_declErr),chunkId, avg(bMagF2) bmf2 from LSST.Object where bMagF > 20.0 GROUP BY chunkId;";
+    std::string expPar = "SELECT sum(pm_declErr) AS QS1_SUM,chunkId,COUNT(bMagF2) AS QS2_COUNT,SUM(bMagF2) AS QS3_SUM FROM LSST.Object_100 AS QST_1_ WHERE bMagF>20.0 GROUP BY chunkId";
```

Comment:
unless you want to test for different casing (which I think you are not doing here), I'd stick with one style, say capitalize SELECT and FROM and WHERE (it is distracting some people if it is unnecessarily different)